### PR TITLE
docs: resubmit `Configuration Files` documentation

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Refers https://github.com/eslint/eslint/issues/17225

Re-submits the changes from https://github.com/eslint/eslint/pull/17068

#### Is there anything you'd like reviewers to focus on?

https://github.com/eslint/eslint/pull/17267/files#r1225040120

### CODE REVIEW

File: docs/src/use/configure/configuration-files-new.md
Comment (snitin315) [inline docs/src/use/configure/configuration-files-new.md:12]: This wording is from the original PR. But shouldn't this be as follows?
```suggestion
This config system feature is complete but not enabled by default. To opt-in, place an `eslint.config.js` file in the root of your project or set the `ESLINT_USE_FLAT_CONFIG` environment variable to `true`. To opt-out, even in the presence of an `eslint.config.js` file, set the environment variable to `false`. If you are using the API, you can use the configuration system described on this page by using the `FlatESLint` class, the `FlatRuleTester` class, or by setting `configType: "flat"` in the `Linter` class.
```

Comment (mdjermanovic) [reply docs/src/use/configure/configuration-files-new.md:12]: I think this was correct. "feature complete" = all planned features are implemented, but it may not be complete (e.g., there could be bugs to fix).

Comment (snitin315) [reply docs/src/use/configure/configuration-files-new.md:12]: Makes sense.

**Review by amareshsm — APPROVED** [review]
LGTM. Waiting for 2nd review.

**Review by mdjermanovic — APPROVED** [review]
The changes LGTM, but I'm not sure if this should have been submitted by the original author of 7162d34df9a66c817c3bd4aafd3a03d226b58dd5 (@nzakas) so I'd like @nzakas to verify.

**Review by nzakas — APPROVED** [review]
As the original author, I give permission for these changes to be merged.

**Pull request merged**
